keel-test/src/main/kotlin/com/netflix/spinnaker/keel/test/Resources.kt

```diff
@@ -0,0 +1,107 @@
+package com.netflix.spinnaker.keel.test
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.netflix.spinnaker.keel.api.ApiVersion
+import com.netflix.spinnaker.keel.api.Monikered
+import com.netflix.spinnaker.keel.api.ResourceSpec
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.ResourceKind
+import com.netflix.spinnaker.keel.api.SPINNAKER_API_V1
+import com.netflix.spinnaker.keel.api.randomUID
+import com.netflix.spinnaker.keel.plugin.ResourceHandler
+import com.netflix.spinnaker.keel.plugin.ResourceNormalizer
+import com.netflix.spinnaker.keel.serialization.configuredObjectMapper
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.util.UUID
+
+val TEST_API = SPINNAKER_API_V1.subApi("test")
+
+fun resource(
+  apiVersion: ApiVersion = TEST_API,
+  kind: String = "whatever",
+  name: String = randomString(),
+  application: String = "fnord"
+): Resource<DummyResourceSpec> =
+  DummyResourceSpec(name = name, application = application)
+    .let { spec ->
+      resource(
+        apiVersion = apiVersion,
+        kind = kind,
+        spec = spec,
+        name = spec.name,
+        application = application
+      )
+    }
+
+fun <T : Monikered> resource(
+  apiVersion: ApiVersion = TEST_API,
+  kind: String = "whatever",
+  spec: T
+): Resource<T> = resource(
+  apiVersion = apiVersion,
+  kind = kind,
+  spec = spec,
+  name = spec.moniker.name,
+  application = spec.application
+)
+
+fun <T : ResourceSpec> resource(
+  apiVersion: ApiVersion = TEST_API,
+  kind: String = "whatever",
+  spec: T,
+  name: String = spec.name,
+  application: String = "fnord"
+): Resource<T> =
```

Comment:
Third variant of test factory, you supply the spec, it (optionally) does the rest. The other variants call this one.